### PR TITLE
8294974: Convert jdk.jshell/jdk.jshell.execution.LocalExecutionControl to use the Classfile API to instrument classes

### PR DESCRIPTION
**Pull request by asotona**
8294974: jdk.jshell jdk.jshell.execution.LocalExecutionControl uses ASM to instrument classes
This patch converts it to use Classfile API.

Please review.
Thanks,
Adam

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294974](https://bugs.openjdk.org/browse/JDK-8294974): Convert jdk.jshell/jdk.jshell.execution.LocalExecutionControl to use the Classfile API to instrument classes

### Reviewers
 * [Jan Lahoda](https://openjdk.org/census#jlahoda) (@lahodaj - **Reviewer**) ⚠️ Review applies to [070be0a8](https://git.openjdk.org/jdk/pull/11413/files/070be0a8a0cf590c4f0e4fba95eb70022ad12d30)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11413/head:pull/11413` \
`$ git checkout pull/11413`

Update a local copy of the PR: \
`$ git checkout pull/11413` \
`$ git pull https://git.openjdk.org/jdk pull/11413/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11413`

View PR using the GUI difftool: \
`$ git pr show -t 11413`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11413.diff">https://git.openjdk.org/jdk/pull/11413.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/module-info.java [hunk 1]
```diff
@@ -381,7 +383,6 @@
         java.prefs;
     exports sun.util.resources to
         jdk.localedata;
-
```
Comment (lahodaj) [inline src/java.base/share/classes/module-info.java:384]: Nit: I'd suggest to preserve the empty line (since there's no other change near).

**Review by lahodaj — APPROVED** [review]
Looks good to me, with a nit in java.base's module-info.

File: src/java.base/share/classes/module-info.java (hunk 1 above)
Comment (asotona) [reply src/java.base/share/classes/module-info.java]: fixed, thanks.

**Pull request closed without merging**